frontend/components/wallet/WalletProvider.tsx

```diff
@@ -25,7 +26,7 @@ interface Props {
  * @returns {JSX.Element} - A provider component that passes down wallet context.
  */
 export const WalletProvider: React.FC<Props> = ({ children }) => {
-  // State to hold the connected account's address.
+  const [provider, setProvider] = useState<any>(null);
```

Comment:
would we not be able to default a provider here? because the current implemention looks as a static pair of constants (based on env variable)